lesson22/main.js

```diff
@@ -145,56 +145,85 @@ const createSortBtn = () => {
   return btn;
 };
 
-const renderSortBtn = () => {
-  const btn = createSortBtn();
-  const targetEle = document.getElementById("id");
-  targetEle.appendChild(btn);
+const setDataSort = (usersData) => {
+  Object.keys(usersData[0]).forEach((key) => {
+    if (targetSortElements.includes(key)) {
+      const element = document.getElementById(key);
+      element.dataset.sort = "sort";
+    }
+  });
+};
+
+const renderSortBtn = (usersData) => {
+  setDataSort(usersData);
+  const elements = document.querySelectorAll("[data-sort]");
+  Array.from(elements).forEach((element) => {
+    const btn = createSortBtn();
+    element.appendChild(btn);
+  });
 };
 
 const clickSortBtn = (usersData) => {
-  const sortArrow = document.querySelector(".sort-btn");
-  const sortArrowImg = document.querySelector(".sort-img");
-  sortArrow.addEventListener("click", () => {
-    changeSortStateAndArrowImg(sortArrowImg);
-    rerenderTableData(usersData);
+  const sortArrowBtns = document.querySelectorAll(".sort-btn");
```

Comment:
```suggestion
  const sortArrowBtns = document.querySelectorAll(".js-sort-btn");
```

Reply:
I fixed it.
[324713d](https://github.com/yuka830/js-lessons/pull/28/commits/324713d61fe85e89052769bbe8c1130035707f4a)